core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/arrayfunction/ArrayAverageExpression.kt

```diff
@@ -2,21 +2,25 @@ package ch.ergon.dope.resolvable.expression.unaliased.type.function.arrayfunctio
 
 import ch.ergon.dope.DopeQuery
 import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.clause.ISelectOffsetClause
 import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.operator.FunctionOperator
 import ch.ergon.dope.validtype.ArrayType
-import ch.ergon.dope.validtype.ValidType
+import ch.ergon.dope.validtype.NumberType
 
-class ArrayAverageExpression<T : ValidType>(
+class ArrayAverageExpression<T : NumberType>(
     private val array: TypeExpression<ArrayType<T>>,
 ) : TypeExpression<T>, FunctionOperator {
     override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
-        val arrayDopeQuery = array.toDopeQuery(manager)
+        val arrayDopeQuery = when (array) {
+            is ISelectOffsetClause<*> -> array.asSubQuery().toDopeQuery(manager)
```

Comment:
I feel this is very couchbase-specific and will be difficult to generalize with mongo.

Reply:
Yes thats true, but supqueries will be difficult to implement in general for mongo